ns/nns.go

```diff
@@ -25,31 +22,9 @@ import (
 type NNS struct {
 	nnsContract util.Uint160
 
-	neoClient neoClient
-}
-
-// represents virtual connection to Neo network used by NNS.Dial.
-type neoClient interface {
-	// calls specified method of the Neo smart contract with provided parameters.
-	call(contract util.Uint160, method string, prm []smartcontract.Parameter) (*result.Invoke, error)
-}
-
-// implements neoClient using Neo HTTP client.
-//
-// note: see NNS.Dial to realize why this isn't defined as type wrapper like neoWebSocket.
-type neoHTTP struct {
-	*rpcclient.Client
-}
-
-func (x *neoHTTP) call(contract util.Uint160, method string, prm []smartcontract.Parameter) (*result.Invoke, error) {
-	return x.Client.InvokeFunction(contract, method, prm, nil)
-}
-
-// implements neoClient using Neo WebSocket client.
-type neoWebSocket rpcclient.WSClient
-
-func (x *neoWebSocket) call(contract util.Uint160, method string, prm []smartcontract.Parameter) (*result.Invoke, error) {
-	return (*rpcclient.WSClient)(x).InvokeFunction(contract, method, prm, nil)
+	invoker interface {
+		Call(util.Uint160, string, ...interface{}) (*result.Invoke, error)
+	}
```

Comment:
It seems unexported interface should have unexported methods

Reply:
why? that is for internal usage only but is implemented by `invoker.Invoker`. first i had just `invoker.Invoker` struct here but forgot about tests that require some interface

Reply:
I thought it would be nice to have the same approach to unexported interfaces across the repository. We already have something similar in pool:
https://github.com/nspcc-dev/neofs-sdk-go/blob/ee92df32032e65d2fb0d8946116d7618c3667bb5/pool/pool.go#L38-L40